src/components/BandInput.js

```diff
@@ -26,7 +26,7 @@ class BandInput extends Component {
         <form onSubmit={(event) => this.handleOnSubmit(event)}>
           <input
             type="text"
-            value={this.state.text}
+            value={this.state.bandName}
```

Comment:
THIS is the part that should be targeted to solution :) And I'd be happy to merge it if we could get these responsibilities isolated into 2 PRs. (single responsibility principle in all things ;) )